modules/shared/src/main/scala/io/constellationnetwork/security/Hash.scala

```diff
@@ -24,12 +24,39 @@ object hash {
   }
 
   object Hash {
+    private val hexDigits = "0123456789abcdef".toCharArray
+    private val sha256 = MessageDigest.getInstance("SHA-256")
 
+    case class Sha256Digest(private val bytes: Array[Byte]) {
+      def toByteArray: Array[Byte] = Array.copyOf(bytes, bytes.length)
+
+      def toHexString: String =
+        bytes
+          .foldLeft(new StringBuilder(bytes.length * 2)) { (sb, b) =>
+            sb.append(hexDigits((b >> 4) & 0xf)).append(hexDigits(b & 0xf))
+          }
+          .toString
+
+      private[constellationnetwork] def toIndexedSeq: IndexedSeq[Byte] = bytes.toIndexedSeq
```

Comment:
Again - why it's private if you open it for the whole package?

Reply:
I made it public again. First, it doesn't cause any issues, and second, it helps save memory for the client if they choose to convert hashes into `Vector`s of bytes.